### DISCUSSION TITLE
Difficulty with STT using FreePBX

### DISCUSSION THREAD

**Discussion by janduy** [deepgram/community#915] [General help] [answered]
Hello to all,

I’m using the project "TeleConnx and Deepgram Bring Speech-to-Text to Asterisk/FreePBX", but I get the message in the log of the service "tcx-dg-stt.service" that no audio was received, below the data of my environment.

Sangoma Linux release 7.8.2003 (Core)
FreePBX Version: 16.0.40.10
Asterisk Version: 18.16.0
Codecs: ulaw

Service status:
```
[root@freepbx ~]# systemctl status tcx-dg-stt.service
● tcx-dg-stt.service - TeleConnx Deepgram STT Service
   Loaded: loaded (/etc/systemd/system/tcx-dg-stt.service; enabled; vendor preset: disabled)
   Active: active (running) since Mon 2024-09-09 11:52:42 UTC; 1s ago
 Main PID: 5679 (tcx-dg-stt)
   CGroup: /system.slice/tcx-dg-stt.service
           └─5679 /usr/local/bin/tcx-dg-stt

Sep 09 11:52:42 freepbx.sangoma.local systemd[1]: tcx-dg-stt.service failed.
Sep 09 11:52:42 freepbx.sangoma.local systemd[1]: Started TeleConnx Deepgram STT Service.
Sep 09 11:52:42 freepbx.sangoma.local tcx-dg-stt[5679]: main: START
Sep 09 11:52:42 freepbx.sangoma.local tcx-dg-stt[5679]: VERSION: 1.0.27.2
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: LICENSE: APPLICATION ID: 31613439-6339-3866-a164-613331373932
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: config: DEEPGRAM: MODEL: nova-2-phonecall.
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: config: DEEPGRAM: LANG: en-US.
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: config: ALERTS: janduyeuclides@gmail.com.
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: config: DEEPGRAM: TRANSCRIPT-NO-FINAL: 180ms.
Sep 09 11:52:43 freepbx.sangoma.local tcx-dg-stt[5679]: socket_created: Service is listening on port 9099.
[root@freepbx ~]# 
```

#When making a connection between internal extensions, the following log message appears
```
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: asterisk_is_connected: T139894624970496: ASTERISK(7): NO AUDIO WAS RECEIVED.
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: MSG Mon Sep  9 11:53:35 2024
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: WARNING: From LSDEV_Stream object.
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: Problem: Received WEBSOCKET_OP_CLOSE message..
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: Occurred in: LSDEV_StreamRawBytesRead().
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: Action: No action taken.
Sep 09 11:53:35 freepbx.sangoma.local tcx-dg-stt[5679]: EOM
```
Example request:
[root@freepbx]# curl -X POST "https://api.deepgram.com/v1/listen" \
> -H "Authorization: Token $API_KEY" \
> --data-binary @sound_test.ulaw

Return:
{"err_code":"Bad Request","err_msg":"Bad Request: failed to process audio: corrupt or unsupported data","request_id":"bf6053f2-f975-492a-b6ce-62a2cdbcdead"}[root@freepbx custom]# 

The uploaded audio has been validated and is working, it is not a corrupted audio.

Can anyone help?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Reply by janduy** [reply]
Example request:
[root@freepbx]# curl -X POST "https://api.deepgram.com/v1/listen" \
> -H "Authorization: Token $API_KEY" \
> --data-binary @sound_test.ulaw

Return:
{"err_code":"Bad Request","err_msg":"Bad Request: failed to process audio: corrupt or unsupported data","request_id":"bf6053f2-f975-492a-b6ce-62a2cdbcdead"}[root@freepbx custom]# 

The uploaded audio has been validated and is working, it is not a corrupted audio.

**Comment by jpvajda** [comment, marked as answer]
@janduy 

I checked our logs and see this request ID had this error:

`Error: Raw binary data was sent to Deepgram without both the encoding and sample_rate set.`

Could you try setting both the encoding and sample_rate in your request?

https://developers.deepgram.com/docs/encoding
https://developers.deepgram.com/docs/sample-rate

**Reply by janduy** [reply]
Hello @jpvajda,
Ok, thank you.
I will check and conduct new tests.

**Comment by gcareri** [comment]
Hi @janduy ,

what you’re trying to build sounds very close to what we’re working on with Agent Voice Response (AVR).
You might want to take a look at AVR — it’s open & free, designed to connect Asterisk or FreePBX with modern ASR (STT), LLM, and TTS providers to create real-time conversational voice bots.

In your case, we already have an integration for Deepgram STT that might fit perfectly:
👉 https://github.com/agentvoiceresponse/avr-sts-deepgram

With AVR you can:
	•	Route calls from Asterisk or FreePBX directly to the AI bot
	•	Use Deepgram for transcription and TTS responses
	•	Plug in an LLM (e.g. Groq, OpenAI, or others) for conversation handling
	•	Support multiple languages (including Spanish)
	•	Build flexible call flows without starting everything from zero

Here’s how to integrate AVR with FreePBX:
👉 https://wiki.agentvoiceresponse.com/en/using-avr-with-freepbx

We’d be happy to help you set this up and adapt it to your use case.
💬 Join our community on Discord for support and guidance: https://discord.gg/MUd3y7eGVF

Regards,
gcareri
